### ISSUE TITLE
sgd optimizer

### ISSUE THREAD

**Issue by wangjiawen2013** [tracel-ai/burn#2359]
Hi,
Ths `sgd` optimizer in burn implements stochastic gradient descent with momentum. Specifically, is it BGD (Batch Gradient Descent) ,SGD (Stochastic Gradient Descent) or MBGD (Mini-Batch Gradient Descent) ?

**Comment by nathanielsimard** [comment]
You can use `sgd` without momemtum. I'm not really familiar with the terms, but it's the same as in PyTorch (https://pytorch.org/docs/stable/generated/torch.optim.SGD.html).

**Comment by wangjiawen2013** [comment]
`SGD` is a bit confusing. Some others people also have the same question with me. And finally we find the answer: 
SGD optimizer in PyTorch actually is Mini-batch Gradient Descent with momentum (MBGD).
https://discuss.pytorch.org/t/how-will-sgd-work-under-batch-data/167252
